### PR TITLE
[lang-c] update to v18

### LINKED ISSUE #1360: update to llvm 18
## Is your feature request related to a problem? Please describe

python JIT require llvm 18

### PR DESCRIPTION
**Pull request by kylos101**
## Description

An alternative to https://github.com/gitpod-io/workspace-images/pull/1375 that'll work in our CI.

## Related Issue(s)

Fixes #1360

## How to test

```console
./build-combo.sh c
```

Tests should pass

## Documentation

/hold

### CODE REVIEW

**Comment by kylos101** [comment]
@gengjiawen I added you as a co-author in 5c71a326e421c2a09929350b095ea42cbc68a4c8

**Comment by kylos101** [comment]
@gengjiawen what did you mean by:

> two new packages was added for the asan support.

**Review by gengjiawen — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 A curated, chronologically ordered list of notable changes in [Gitpod's default workspace images](https://hub.docker.com/u/gitpod).
 
+## 2024-07-01
+
+- Bump Clang to `18`
```
Comment (gengjiawen) [inline CHANGELOG.md:7]:
```suggestion
- Bump llvm to 18, and enable ASAN support.
```

Comment (kylos101) [reply CHANGELOG.md:7]: For consistency, I'd like to keep using CLang. [[1](https://github.com/gitpod-io/workspace-images/blob/5c71a326e421c2a09929350b095ea42cbc68a4c8/CHANGELOG.md#L80)][[2](https://hub.docker.com/r/gitpod/workspace-c)]

But, I will add ` and enable ASAN support`. Thanks!

File: tests/lang-c.yaml [hunk 2]
```diff
@@ -12,3 +12,9 @@
   command: [wc,/etc/apt/sources.list.d/llvm.list]
   assert:
   - status == 0
+- desc: it should have major version
+  command: [npx envinfo]
```
Comment (gengjiawen) [inline tests/lang-c.yaml:16]: npx comes from Node.js, looks not available on lang-c. Maybe use `clang --version`

Comment (kylos101) [reply tests/lang-c.yaml:16]: Aye, on it!

Comment (kylos101) [reply tests/lang-c.yaml]: Fixed in [c767edd](https://github.com/gitpod-io/workspace-images/pull/1377/commits/c767edd446fdd058f0d9a4e0d313bf9676a711fb)

**Comment by gengjiawen** [comment]
Looks CI passed. Let's merge this.

**Review by mustard-mh — APPROVED** [review]
To unblock

**Pull request merged**